api/src/controllers/team.js

```diff
@@ -313,6 +316,75 @@ async function updateTeam(req, res, next) {
   }
 }
 
+/**
+  * Clear the team Data from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: teamn is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function resetTeamData(req, res, next) {
+  try {
+    const { teamId } = req.params;
+    const { isDeleteFlag } = req.body;
+    await Achievement.deleteMany({ teamId });
+    await Publication.deleteMany({ teamId });
+    if (isDeleteFlag) {
+      await Team.findByIdAndDelete(teamId);
+      return res.status(200).json('Deleted successfully!');
+    } return res.status(200).json('Cleared successfully!');
+  } catch (error) {
+    return next(
+      fillErrorObject(500, 'Error occurred with server', [error.message]),
+    );
+  }
+}
+
+/**
+  * Delete the team from the database on /team/:team_id
+  * @param {} req request object, containing team id in the url
+  * @param {*} res response object, the deleted team document
+  * @returns 200: team is deleted
+  * @returns 404: team is not found
+  * @returns 400: team id is not in a valid hexadecimal format
+  */
+async function deleteGHPages(req, res, next) {
+  const { ghToken } = req.body;
+  // Call github API to get username
+  const { data } = await axios.get('https://api.github.com/user',
+    {
+      headers: { Authorization: `token ${ghToken}` },
+    });
+
+  // Creating repoName
+  const ghUsername = data.login;
+  logger.info(`GitHub Pages delete initiated for user: ${ghUsername}`);
+  const repoName = `${ghUsername}.github.io`;
+  // delete repo
+  try {
+    const deleteRepo = await axios.delete(
+      `https://api.github.com/repos/${ghUsername}/${repoName}`,
+      {
+        headers: {
+          Authorization: `token ${ghToken}`,
+          Accept: 'application/vnd.github.v3+json',
+        },
+      },
+    );
+    if (deleteRepo.status === 204) {
+      logger.info(`GitHub pages deleted for user: ${ghUsername}`);
+    }
+    // result logged
+    return res.status(200).json('Deleted successfully!');
+  } catch (error) {
+    logger.info(` Failed: GitHub pages not deleted for user: ${ghUsername}!`);
+    return next(
+      fillErrorObject(500, 'Error occurred with server!', [error.message]),
```

Comment:
The error message that shows up cause of this line is: `Request failed with status code 403` and might not be too user friendly. Might be worth changing to something like 'Failed to delete GitHub pages'.